lib/cc/analyzer/engines_builder.rb

```diff
@@ -44,7 +44,7 @@ def engine_config(raw_engine_config)
 
       def names_and_raw_engine_configs
         {}.tap do |ret|
-          @config.engines.each do |name, raw_engine_config|
+          (@config.engines || []).each do |name, raw_engine_config|
```

Comment:
Just tried this locally, and it broke… a lot of things. `Array.wrap` & YAML `Node`s both have some surprising behavior, and they do not play well together.

@GordonDiggs do you have other concerns around this PR?